comm/src/main/scala/coop/rchain/comm/transport/GrpcTransportClient.scala

```diff
@@ -78,26 +72,34 @@ class GrpcTransportClient(
         case Left(t)           => Task.raiseError[SslContext](t)
       }
 
-  private def createChannel(peer: PeerNode): Task[ManagedChannel] =
+  private def createChannel(peer: PeerNode): Task[BufferedGrpcStreamChannel] =
     for {
       clientSslContext <- clientSslContextTask
-      c <- Task.delay {
-            NettyChannelBuilder
-              .forAddress(peer.endpoint.host, peer.endpoint.tcpPort)
-              .executor(scheduler)
-              .maxInboundMessageSize(maxMessageSize)
-              .negotiationType(NegotiationType.TLS)
-              .sslContext(clientSslContext)
-              .intercept(new SslSessionClientInterceptor(networkId))
-              .overrideAuthority(peer.id.toString)
-              .build()
-          }
-    } yield c
-
-  def getChannel(peer: PeerNode): Task[ManagedChannel] =
+      grpcChannel = NettyChannelBuilder
+        .forAddress(peer.endpoint.host, peer.endpoint.tcpPort)
+        .executor(scheduler)
+        .maxInboundMessageSize(maxMessageSize)
+        .negotiationType(NegotiationType.TLS)
+        .sslContext(clientSslContext)
+        .intercept(new SslSessionClientInterceptor(networkId))
+        .overrideAuthority(peer.id.toString)
+        .build()
+      buffer = new StreamObservable(peer, clientQueueSize, tempFolder)
+
+      channel = BufferedGrpcStreamChannel(grpcChannel, buffer)
+
+      _ = buffer.subscribe(
+        sMsg =>
+          streamBlobFile(sMsg.path, peer, sMsg.sender)
+            .guarantee(sMsg.path.deleteSingleFile[Task])
+            .runToFuture >> Continue.pure[CancelableFuture]
+      )
+    } yield channel
```

Comment:
Please add log message when new channel is created.

Reply:
@tgrospic here is it 
https://github.com/rchain/rchain/blob/8190cf547f03de903c8d6699e7842f23c7279151/comm/src/main/scala/coop/rchain/comm/transport/GrpcTransportClient.scala#L111-L113

Reply:
I moved to `createChannel` method, thanks for hint.